Feature/refactor build

steveblue opened the pull request:
TY @jasonaden for the talk at ng-conf. This branch is almost ready to roll. Just gotta inline templates and CSS and then DONE. `npm run build:lib` will run the build for a library.

jasonaden commented on build.dev.js:
```diff
@@ -0,0 +1,430 @@
+"use strict";
```
This file plus build.lib.js and build.prod.js seem to have quite a bit of the same code. Like `log` and `warn` and others. Maybe you can create a build.utils.js and just require the utils into these 3 files. It would make it a lot easier to read through the build for the important pieces.

steveblue replied:
Yeah, thats going to be part of the refactor for sure, I need to clean up build.lib.js.

steveblue replied:
Cleaned!

jasonaden commented on dist/index.ts:
```diff
@@ -0,0 +1 @@
+export * from './default/default.module';
```
This file shouldn't be published.

jasonaden commented on src/app/shared/components/lib/default/default.module.ts:
```diff
@@ -0,0 +1,11 @@
+@NgModule({
+  imports:      [ CommonModule ],
+  declarations: [ DefaultComponent ]
+})
```
Add `exports: [ DefaultComponent]` so this component can be used in the application.

steveblue replied:
I'm also looking into the code for `rollup-plugin-angular` and will probably use a variant of it to parse the `@Component` and the inline the template and style prior to `ngc`.

jasonaden commented on src/app/shared/components/lib/default/default.module.ts:
```diff
@@ -0,0 +1,11 @@
+import { CommonModule }  from '@angular/common';
+
+import { DefaultComponent } from './default.component';
+
```
`export * from ./default.component`

This will re-export the component itself so your app developers can import it and use it in their code.

steveblue commented:
@jasonaden Is the current build in `./dist` what you had in mind at ng-conf for optimizing the build with inline CSS and HTML? TY for your comments!

steveblue commented:
It looks like there's template strings in the es5 build of the library. That shouldn't be the case probably.

steveblue commented:
- Fixed inline templates and CSS for library build
- Fixed livereload in dev and production builds when SCSS files where changed
- Updated README
- Added UMD module pattern to library build
